<a href="https://colab.research.google.com/github/yuniarachmaa/TUGAS-MATEMATIKA-DISKRIT-A/blob/main/MATDIS_TUGAS_3.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nama : Yuniar Rachmawati

NPM : 24083010106

Mata Kuliah : Matematika Diskrit (A)

# TUGAS MATEMATIKA DISKRIT 3

In [ ]:
def hitung_harmonik(n):
    total = 0
    for i in range(1, n + 1):
        total += 1 / i
    return total

n = 1
while True:
    jumlah = hitung_harmonik(n)
    print(f"n = {n}, H_n = {jumlah:.6f}")
    if jumlah > 2:
        print(f"--> Pertama kali lebih dari 2 terjadi pada n = {n}")
        break
    n += 1

n = 1, H_n = 1.000000
n = 2, H_n = 1.500000
n = 3, H_n = 1.833333
n = 4, H_n = 2.083333
--> Pertama kali lebih dari 2 terjadi pada n = 4


Program yang dituliskan bertujuan untuk mencari nilai 𝑛 pertama kali ketika jumlah deret harmonik 𝐻𝑛 lebih besar dari 2. Fungsi hitung_harmonik(n) dibuat untuk menghitung jumlah harmonik sampai suku ke-𝑛. Selanjutnya, dilakukan perulangan mulai dari 𝑛 = 1 dan setiap hasil perhitungan 𝐻𝑛 ditampilkan.

Berdasarkan hasil eksekusi, diperoleh urutan nilai:
* Saat 𝑛 = 1, jumlah harmonik 𝐻1 = 1.000000
* Saat 𝑛 = 2, jumlah harmonik 𝐻2 = 1.500000
* Saat 𝑛 = 3, jumlah harmonik 𝐻3 = 1.833333
* Saat 𝑛 = 4, jumlah harmonik 𝐻4 = 2.083333

Dari data tersebut dapat dilihat bahwa deret harmonik mengalami pertambahan nilai setiap kali
𝑛 ditambah. Pertama kali jumlah harmonik melampaui angka 2 adalah pada 𝑛 = 4 dengan hasil 𝐻4 = 2.083333
Dengan demikian, program ini berhasil menunjukkan bahwa nilai ambang 2 pada deret harmonik terlampaui ketika 𝑛 = 4.